frontend/lib/start-account-or-login/verify-password.tsx

```diff
@@ -85,7 +99,18 @@ export const VerifyPassword: React.FC<StartAccountOrLoginProps> = ({
       >
         {(ctx) => (
           <>
-            <HiddenFormField {...ctx.fieldPropsFor("phoneNumber")} />
+            {session.lastQueriedPhoneNumber ? (
+              <HiddenFormField {...ctx.fieldPropsFor("phoneNumber")} />
+            ) : (
+              <>
+                <br />
+                <PhoneNumberFormField
+                  {...ctx.fieldPropsFor("phoneNumber")}
+                  label={li18n._(t`Phone number`)}
+                />
+              </>
+            )}
```

Comment:
This is not exactly DRY but given all of the referenced parts like ctx and session, it ended up adding a lot more complexity to factor this component out... I think it's ok to leave like this... right??